TF-M PSA template: Lock Approtect in network core (#17093)
nrfconnect/sdk-nrf: samples/tfm/tfm_psa_template/README.rst

It includes provisioning the device with keys and being able to perform a device firmware update.
The sample prints information about the identity of the device and the firmware versions that are currently running.

> With nRF5340 this sample also includes the :ref:`B0n bootloader <nc_bootloader>` and :ref:`empty_net_core <nrf5340_empty_net_core>` for demonstrating the network core firmware update process.

Reviewer: ```suggestion
On the nRF5340 devices, this sample also includes the :ref:`B0n bootloader <nc_bootloader>` and the :ref:`empty_net_core <nrf5340_empty_net_core>` image for demonstrating the network core firmware update process.
```

    mcumgr --conntype serial --connstring dev=/dev/ttyACM1,baud=115200,mtu=512 reset

Network core update (nRF5340 only)
==================================

To upload a new network core image, build the empty_net_core image with an updated firmware image version.

.. code-block:: console

    west build -b nrf5340dk/nrf5340/cpuapp/ns nrf/samples/tfm/tfm_psa_template -d build_update \
    -Dempty_net_core_CONFIG_FW_INFO_FIRMWARE_VERSION=2

> Then upload the new network core image to the device. Note that the image is uploaded to the network core slot.

Reviewer: ```suggestion
Then upload the new network core image to the device.
Note that the image is uploaded to the network core slot.
```

.. code-block:: console

    mcumgr --conntype serial --connstring dev=/dev/ttyACM1,baud=115200,mtu=512 image test <hash>

Trigger the network core update by initiating a reset.
The verification of the image will happen during the update process.

.. code-block:: console

    mcumgr --conntype serial --connstring dev=/dev/ttyACM1,baud=115200,mtu=512 reset

Alternatively, you can conduct a manual reset to trigger the network core update.
This allows you to observe the update process in the application and network core console outputs.

Simultaneous application and network core update (nRF5340 only)
===============================================================

When the interface between the application and network core is updated, both the application and network core images must be updated simultaneously.
To do this, build the application image with an updated image version and the network core image with an updated firmware image version.

.. code-block:: console

    west build -b nrf5340dk/nrf5340/cpuapp/ns nrf/samples/tfm/tfm_psa_template -d build_update \
    -DCONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION=\"1.2.4\" -Dempty_net_core_CONFIG_FW_INFO_FIRMWARE_VERSION=3

> Then upload the new application and network core images to the device. Note that the application image is uploaded to the application slot and the network core image is uploaded to the network core slot.

Reviewer: ```suggestion
Then upload the new application and network core images to the device.
Note that the application image is uploaded to the application slot, and the network core image is uploaded to the network core slot.
```

.. code-block:: console

    mcumgr --conntype serial --connstring dev=/dev/ttyACM1,baud=115200,mtu=512 image upload \
    build_update/tfm_psa_template/zephyr/zephyr.signed.bin -e -n 0

    mcumgr --conntype serial --connstring dev=/dev/ttyACM1,baud=115200,mtu=512 image upload \
    build_update/signed_by_mcuboot_and_b0_empty_net_core.bin -e -n 1

    mcumgr --conntype serial --connstring dev=/dev/ttyACM1,baud=115200,mtu=512 image list

> Once the images are uploaded, the hash of the images is shown in the image list. The application image is image 1 in slot 0 and the network core image is image 1 in slot 1.
> To allow the application and network core images to be updated simultaneously, first confirm the network core image and then the application image.

Reviewer: ```suggestion
Once the images are uploaded, the hash of the images is shown in the image list.
The application image is image 1 in slot 0, and the network core image is image 1 in slot 1.
To allow the application and network core images to be updated simultaneously, first confirm the network core image and then the application image.
```

.. code-block:: console

    mcumgr --conntype serial --connstring dev=/dev/ttyACM1,baud=115200,mtu=512 image confirm <network core image hash>

    mcumgr --conntype serial --connstring dev=/dev/ttyACM1,baud=115200,mtu=512 image confirm <application core image hash>

Trigger the core updates by initiating a reset.
The verification of the images will happen during the update process.

.. code-block:: console

    mcumgr --conntype serial --connstring dev=/dev/ttyACM1,baud=115200,mtu=512 reset

Alternatively, you can conduct a manual reset to trigger the core updates.
This allows you to observe the update process in the application and network core console outputs.

Dependencies
*************

* This sample uses the TF-M module found in the :file:`modules/tee/tfm/` folder of the |NCS|.
* This sample uses the :ref:`lib_tfm_ioctl_api` library.

> * This sample uses the :ref:`subsys_pcd` library with nRF5340.

Reviewer: ```suggestion
* On the nRF5340 devices, this sample uses the :ref:`subsys_pcd` library.
```
